internal/driver/glfw/window_test.go

```diff
@@ -527,15 +532,51 @@ func TestWindow_TappedIgnoredWhenMovedOffOfTappable(t *testing.T) {
 	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Release, 0)
 
 	w.waitForEvents()
+
 	assert.Equal(t, 0, tapped, "button was tapped without mouse press & release on it %d", tapped)
 
 	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Press, 0)
 	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Release, 0)
 
 	w.waitForEvents()
+
 	assert.Equal(t, 2, tapped, "Button 2 should be tapped")
 }
 
+func TestWindow_TappedAndDoubleTapped(t *testing.T) {
+	w := createWindow("Test").(*window)
+	tapped := 0
+	but := newDoubleTappableButton()
+	but.OnTapped = func() {
+		tapped = 1
+	}
+	but.onDoubleTap = func() {
+		tapped = 2
+	}
+	w.SetContent(fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, nil, nil, nil), but))
+
+	w.mouseMoved(w.viewport, 15, 25)
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Press, 0)
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Release, 0)
+
+	w.waitForEvents()
+	time.Sleep(500 * time.Millisecond)
+
+	assert.Equal(t, 1, tapped, "Single tap should have fired")
+	tapped = 0
+
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Press, 0)
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Release, 0)
+	w.waitForEvents()
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Press, 0)
+	w.mouseClicked(w.viewport, glfw.MouseButton1, glfw.Release, 0)
```

Comment:
could also use test.DoubleTap now :)

Reply:
I could - but that doesn't actually simulate mouse clicks from what I can tell, it just calls the doubletapped function on an object.  This actually does the mouse clicks so feels more solid?

Reply:
Ack